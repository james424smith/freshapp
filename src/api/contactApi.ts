import { Contact, SuccessRestReturn } from "../interfaces";
import contactDetails from "../../__mocks__/fake-data/contactApi.json";

const delay = 0;

const getContactDetails = (): Promise<SuccessRestReturn<Contact>> => {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve(Object.assign({}, { json: () => contactDetails }));
    }, delay);
  });
};

export default getContactDetails;
