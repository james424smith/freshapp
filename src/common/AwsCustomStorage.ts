import Keychain from "react-native-secure-storage";
const memoryKeyPrefix = "@MemoryStorage:";

interface MemoryData {
  [key: string]: string;
}
async function handleItemInKeychain(
  key: string,
  method: string,
  value?: string
) {
  try {
    if (method === "get") {
      return Keychain.getItem(key);
    } else if (method === "remove") {
      return Keychain.removeItem(key);
    } else {
      return Keychain.setItem(key, value);
    }
  } catch (error) {
    throw Error(error);
  }
}

export default class AmplifyCommunityAsyncStorage {
  static memoryData: MemoryData = {};

  static multiGet = async function (keys: string[], type: string) {
    return Promise.all(
      keys.map(async (key) => {
        await new Promise((res) => setTimeout(res, 100)); // sleep for 100ms
        return [
          key,
          await handleItemInKeychain(key, type).catch((error) =>
            console.log(`Error while trying ${type} for ${key}`, error)
          ),
        ];
      })
    );
  };

  static syncPromise: void | undefined;
  static setItem = async (key: string, value: string) => {
    try {
      await handleItemInKeychain(`${memoryKeyPrefix}${key}`, "set", value);
    } catch (error) {
      console.info(`Failed to set item ${key} in AsyncStorage: ${error}`);
    }
    AmplifyCommunityAsyncStorage.memoryData[key] = value;
    return value;
  };
  static getItem = (key: string) => {
    const value = AmplifyCommunityAsyncStorage.memoryData[key];
    return value === undefined ? null : value;
  };

  static removeItem = async (key: string) => {
    try {
      await handleItemInKeychain(`${memoryKeyPrefix}${key}`, "remove");
      delete AmplifyCommunityAsyncStorage.memoryData[key];
    } catch (error) {
      console.info(`Failed to remove item ${key} in AsyncStorage: ${error}`);
    }
  };
  static clear = async () => {
    const keys = await Keychain.getAllKeys();
    const memoryKeys = keys.filter((key: string) =>
      key.startsWith(memoryKeyPrefix)
    );
    await AmplifyCommunityAsyncStorage.multiGet(memoryKeys, "remove");
    AmplifyCommunityAsyncStorage.memoryData = {};
  };
  static createSyncPromise = async (_temp?: string) => {
    AmplifyCommunityAsyncStorage.memoryData = {};
    const keys = await Keychain.getAllKeys();
    const memoryKeys = keys.filter((key: string) =>
      key.startsWith(memoryKeyPrefix)
    );
    const pairs = await AmplifyCommunityAsyncStorage.multiGet(
      memoryKeys,
      "get"
    );
    pairs.forEach(([memoryKey, value]) => {
      if (value) {
        const key = (memoryKey as string).replace(memoryKeyPrefix, "");
        AmplifyCommunityAsyncStorage.memoryData[key] = value;
      }
    });
  };

  static sync = async () => {
    console.log(AmplifyCommunityAsyncStorage.syncPromise);
    if (!AmplifyCommunityAsyncStorage.syncPromise) {
      AmplifyCommunityAsyncStorage.syncPromise = await AmplifyCommunityAsyncStorage.createSyncPromise();
    }
    return AmplifyCommunityAsyncStorage.syncPromise;
  };
}
