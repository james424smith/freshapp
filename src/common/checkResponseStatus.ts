export const checkResponseStatus = (error: { status: number }) => {
  if (error.status === 404) {
    return [];
  } else {
    throw error;
  }
};
