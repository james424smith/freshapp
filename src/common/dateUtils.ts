import { DateTime, DurationObject } from "luxon";
import moment from "moment-timezone";

export type DateObject = {
  day: number;
  month: number;
  year: number;
};

export const getTimestampFromDate = (date?: string): number =>
  date
    ? DateTime.fromFormat(date, "dd-MM-yyyy", { zone: "UTC" }).toMillis()
    : 0;

export const timeDiffTextMillis = (fromDate?: number, toDate?: number) => {
  if (fromDate && toDate) {
    const luxonFrom = DateTime.fromMillis(fromDate, { zone: "UTC" });
    const luxonTo = DateTime.fromMillis(toDate, { zone: "UTC" });
    const diff: DurationObject = luxonTo
      .diff(luxonFrom, ["days", "hours", "minutes", "seconds"])
      .toObject();

    if (diff.days && diff.days > 0) {
      if (diff.days && diff.days < 7) {
        return `${Math.ceil(Number(diff.days))}d`;
      } else {
        return luxonFrom.toFormat("dd.MM.yy");
      }
    } else if (diff.hours && diff.hours > 0) {
      return `${Math.ceil(Number(diff.hours))}h`;
    } else if (diff.minutes && diff.minutes > 0) {
      return `${Math.ceil(Number(diff.minutes))}m`;
    } else {
      return `1m`;
    }
  }
  throw Error("fromDate and toDate must both be defined");
};

export const formatDate = (date?: Date, format = "dd.MM.yyyy") =>
  date ? DateTime.fromJSDate(date).toFormat(format) : "";

export const timeToNow = (fromDate?: number, toDate?: number) => {
  if (fromDate && toDate) {
    const luxonFrom = DateTime.fromMillis(fromDate, { zone: "UTC" });
    const luxonTo = DateTime.fromMillis(toDate, { zone: "UTC" });
    const diff: DurationObject = luxonTo
      .diff(luxonFrom, ["hours", "minutes"])
      .toObject();

    return diff.minutes === 0 ? `${diff.hours}h` : `<${(diff.hours || 0) + 1}h`;
  } else {
    throw Error("fromDate and toDate must both be defined");
  }
};
export const timestampToUtcString = (timestamp?: number) => {
  if (timestamp) {
    const date = DateTime.fromMillis(timestamp, { zone: "UTC" });
    return date.toFormat("dd.MM.yy HH:mm");
  } else {
    throw Error("timestamp must be defined");
  }
};

export const timestampToLocalString = (timestamp?: number) => {
  if (timestamp) {
    const date = DateTime.fromMillis(timestamp, { zone: "UTC" }).toLocal();
    return date.toFormat("dd.MM.yy    HH:mm");
  } else {
    throw Error("timestamp must be defined");
  }
};

export const compareToDateFromDateStartOfDay = (toDate: DateObject) => {
  const { year, month, day } = toDate;
  /**
   * Month starts from 00. so January is 00, February 01, etc.
   */
  if (year && month && day) {
    return (
      new Date().getTime() >=
      new Date(toDate.year, toDate.month - 1, toDate.day).getTime()
    );
  } else {
    throw Error("year, month, and date must all be defined");
  }
};

export const getTimeFromTimezone = (
  dateTime: Date,
  timeZone?: string,
  format = "dd.MM.yyyy HH:mm"
) => {
  if (timeZone) {
    return moment(dateTime, format).tz(timeZone).format(format);
  } else {
    return moment(dateTime, format).format(format);
  }
};
