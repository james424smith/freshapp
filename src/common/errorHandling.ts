export default (errorCode?: string) => {
  switch (errorCode) {
    case "NetworkError":
      return "You are offline. Please check your internet connection.";
    case "signUpOracle":
      return "Email address entered does not match our records. In order to proceed with registration please either enter a different email address or contact our Support Team (feedback.app@marlowgroup.com) in order to update your records. Please always include your Marlow ID when contacting our Support Team.";
    case "signUpAws":
      return "There is a problem. Seems you already started or completed the registration process. You need to go back to the main screen and login with your Marlow ID and password. If you did not create a personal password yet then please check your email account for the temp password.";
    case "UserNotFoundException":
      return "Please make sure your Employee Id and/or password are correct.";
    case "Login":
      return "There is a problem. Did you already create an account? If not, please select the 'Create New Account' option and then copy / paste the password from the email. If you already created an account please ensure you are entering the correct Marlow ID and personal password. If you don't remember your personal password please use the 'Forgot Password' option";
    case "ForgotPassword":
      return "A verification code has been sent to the registered email of the provided Employee ID. Please use this code to reset your password.";
    default:
      return "There is a problem. Did you already create an account? If not, please select the 'Create New Account' option and then copy / paste the password from the email. If you already created an account please ensure you are entering the correct Marlow ID and personal password. If you don't remember your personal password please use the 'Forgot Password' option";
  }
};
