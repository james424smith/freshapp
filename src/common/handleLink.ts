import { Linking } from "react-native";

export const handleLink = (
  redirectType: string,
  destination?: string
): Promise<void> | "" | undefined => {
  if (redirectType === "url") {
    return destination && Linking.openURL(destination);
  } else if (redirectType === "crewcompanion") {
    return destination && Linking.openURL(`crewcompanion://${destination}`);
  } else {
    return destination && Linking.openURL(`${redirectType}:${destination}`);
  }
};
