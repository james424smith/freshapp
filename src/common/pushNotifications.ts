import { Auth } from "aws-amplify";
import * as RNFS from "react-native-fs";
import AsyncStorage from "@react-native-community/async-storage";
import { SIGN_IN_ROUTE } from "../constants/routes";
import { AppState, Platform, Vibration, PushNotification } from "react-native";
import PushNotificationIOS from "@react-native-community/push-notification-ios";
import { ANDROID, IOS } from "../constants/Platforms";
import {
  ASSIGNMENT,
  CHAT,
  FLIGHT,
  GENERAL,
  NEWSLETTER,
  PAYSLIP,
  PERSONAL,
  DELETE_USER,
  LOGOUT,
} from "../constants/notificationsConstants";
import * as PushNotificationsHelper from "./pushNotifications";
import PushNotificationAmplify from "@aws-amplify/pushnotification";

import {
  SET_NEW_CHAT_NOTIFICATION,
  PushNotificationType,
  InitialisedSagaCall,
  WebsocketEvent,
} from "../interfaces";
import { handleLink } from "./handleLink";

const errorMessage = "error while redirecting to screen";

export const handleSignOut = async (shouldBeGlobal: boolean) => {
  const currentEmployeeId = await AsyncStorage.getItem("employeeId");
  const shouldSaveEmployeeId = await AsyncStorage.getItem(
    "shouldSaveEmployeeId"
  );

  await Auth.signOut({ global: shouldBeGlobal });
  const documentPath = `${RNFS.DocumentDirectoryPath}/document`;
  const documentFolder = await RNFS.exists(documentPath);
  documentFolder && (await RNFS.unlink(documentPath));
  const payslipPath = `${RNFS.DocumentDirectoryPath}/payslip`;
  const payslipFolder = await RNFS.exists(payslipPath);
  payslipFolder && (await RNFS.unlink(payslipPath));
  const asyncStorageKeys = await AsyncStorage.getAllKeys();
  if (asyncStorageKeys.length > 0) {
    Promise.all(
      asyncStorageKeys.map(async (key: string) => {
        if (key !== "colorScheme" && key !== "SIGN_IN_PATH") {
          await AsyncStorage.removeItem(key);
        }
      })
    );
  }

  currentEmployeeId &&
    (await AsyncStorage.setItem("employeeId", currentEmployeeId));
  shouldSaveEmployeeId &&
    (await AsyncStorage.setItem("shouldSaveEmployeeId", shouldSaveEmployeeId));
};

type Props = {
  getNotificationsDetails: () => InitialisedSagaCall;
  dispatch: (arg?: unknown) => void;
  navigation: any;
  setWebsocketData: (event: WebsocketEvent) => InitialisedSagaCall;
};

export const extractNotificationData = (notification: PushNotificationType) => {
  if (Platform.OS === ANDROID) {
    return notification.data
      ? JSON.parse(notification.data["pinpoint.jsonBody"])
      : JSON.parse(notification["pinpoint.jsonBody"]);
  } else {
    return notification._data.data.jsonBody;
  }
};

export const extractNotificationBody = (notification: PushNotificationType) => {
  if (notification.body) {
    return notification.body;
  } else {
    return notification._alert?.body;
  }
};

export const handlePushNotificationRedirection = async ({
  notificationCategory,
}: {
  notificationCategory: string;
}) => {
  switch (notificationCategory) {
    case DELETE_USER:
      await PushNotificationsHelper.handleSignOut(true);
      break;
    case LOGOUT:
      await PushNotificationsHelper.handleSignOut(false);
      break;
    case ASSIGNMENT:
      await handleLink("crewcompanion", "assignment");
      break;
    case PAYSLIP:
      await handleLink("crewcompanion", "salaries");
      break;
    case FLIGHT:
      await handleLink("crewcompanion", "flights");
      break;
    case CHAT:
      await handleLink("crewcompanion", "chat/history");
      break;
    case PERSONAL:
      await handleLink("crewcompanion", "documents");
      break;
    case NEWSLETTER:
      await handleLink("crewcompanion", "news");
      break;
    case GENERAL:
      await handleLink("crewcompanion", "notifications");
      break;
    default:
      await handleLink("crewcompanion", "notifications");
      break;
  }
};

export default async () => {
  // // required on iOS only (see fetchCompletionHandler docs: https://facebook.github.io/react-native/docs/pushnotificationios.html)
  if (Platform.OS === IOS) {
    PushNotificationIOS.getInitialNotification().then(
      (notification: PushNotification | null) => {
        if (notification !== null) {
          const notificationData = PushNotificationsHelper.extractNotificationData(
            (notification as unknown) as PushNotificationType
          );
          PushNotificationsHelper.handlePushNotificationRedirection(
            notificationData
          )
            .then(() => {})
            .catch((error) => console.log(errorMessage, error));
          // required on iOS only (see fetchCompletionHandler docs: https://facebook.github.io/react-native/docs/pushnotificationios.html)
          notification.finish(PushNotificationIOS.FetchResult.NoData);
        }
      }
    );
  }
  // get the notification data when notification is received
  PushNotificationAmplify.onNotification(
    async (notification: PushNotification | null) => {
      Vibration.vibrate([0, 500, 100, 200]);
      const notificationData = PushNotificationsHelper.extractNotificationData(
        (notification as unknown) as PushNotificationType
      );
      if (Platform.OS === ANDROID) {
        if (notificationData.notificationCategory === "DeleteUser") {
          PushNotificationsHelper.handlePushNotificationRedirection(
            notificationData
          ).catch((error) => console.log(errorMessage, error));
        }
      } else {
        if (AppState.currentState === "background" && notification) {
          PushNotificationsHelper.handlePushNotificationRedirection(
            notificationData
          ).catch((error) => console.log(errorMessage, error));
          // required on iOS only (see fetchCompletionHandler docs: https://facebook.github.io/react-native/docs/pushnotificationios.html)
          notification.finish(PushNotificationIOS.FetchResult.NoData);
        }
      }
    }
  );

  // get the notification data when notification is opened
  PushNotificationAmplify.onNotificationOpened(
    async (notification: PushNotification | null) => {
      const notificationData = PushNotificationsHelper.extractNotificationData(
        (notification as unknown) as PushNotificationType
      );
      await PushNotificationsHelper.handlePushNotificationRedirection(
        notificationData
      ).catch((error) => console.log(errorMessage, error));
    }
  );
};

export const getNotificationOnForeground = async (props: Props) =>
  PushNotificationAmplify.onNotification(
    async (notification: PushNotification | null) => {
      Vibration.vibrate([0, 500, 100]);
      console.log("in app notification", notification);
      props.getNotificationsDetails();
      const notificationData = PushNotificationsHelper.extractNotificationData(
        (notification as unknown) as PushNotificationType
      );
      if (notificationData.notificationCategory === DELETE_USER) {
        await PushNotificationsHelper.handleSignOut(true).then(() => {
          props.navigation.navigate("Auth", { screen: SIGN_IN_ROUTE });
        });
      }
      if (notificationData.notificationCategory === LOGOUT) {
        await PushNotificationsHelper.handleSignOut(false).then(() =>
          props.navigation.navigate("Auth", { screen: SIGN_IN_ROUTE })
        );
      }
      if (notificationData.notificationCategory === CHAT) {
        const event = {
          messageType: SET_NEW_CHAT_NOTIFICATION,
          payload: {
            ...notificationData,
            notificationBody: PushNotificationsHelper.extractNotificationBody(
              (notification as unknown) as PushNotificationType
            ),
          },
        };
        props.dispatch(props.setWebsocketData(event as WebsocketEvent));
      }
    }
  );
