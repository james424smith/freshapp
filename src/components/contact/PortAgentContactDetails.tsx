import React from "react";
import { View, ScrollView, TouchableOpacity } from "react-native";
import Text from "../StyledText";
import styles from "../../styles/more_screens/ManningAgentStyles";
import { PortAgent } from "../../interfaces";
import { handleLink } from "../../common/handleLink";
import { useTheme, Theme } from "@react-navigation/native";
import valueOrDefault from "../../common/valueOrDefault";
import { WebsiteView } from "./CrewManagerContactDetails";
type Props = { portAgentDetails?: PortAgent };

const renderPortAgentDetails = (theme: Theme, portAgent?: PortAgent) => {
  return [
    <View>
      <Text style={{ ...styles.detailsLabel, color: theme.colors.text }}>
        Port Agent Name
      </Text>
      <Text style={{ ...styles.detailsValue, color: theme.colors.text }}>
        {valueOrDefault(portAgent?.name, "-")}
      </Text>
    </View>,
    <View style={styles.numbersView}>
      <View style={styles.columnContainer}>
        <Text style={{ ...styles.detailsLabel, color: theme.colors.text }}>
          Direct Phone Numbers
        </Text>
        {(valueOrDefault(
          portAgent?.directTelephoneNumbers,
          []
        ) as string[]).map((directTelephoneNumber: string, i: number) => (
          <TouchableOpacity
            key={i}
            testID="directTelephoneNumber-button"
            onPress={() => handleLink("tel", directTelephoneNumber)}
          >
            <Text style={{ ...styles.detailsValueU, color: theme.colors.text }}>
              {valueOrDefault(directTelephoneNumber, "")}
            </Text>
          </TouchableOpacity>
        ))}
      </View>
      <View style={styles.columnContainer}>
        <Text style={{ ...styles.detailsLabel, color: theme.colors.text }}>
          AOH Phone Numbers
        </Text>
        {(valueOrDefault(portAgent?.aohTelephoneNumbers, []) as string[]).map(
          (aohTelephoneNumber: string, i: number) => (
            <TouchableOpacity
              key={i}
              testID="aohTelephoneNumber-button"
              onPress={() => handleLink("tel", aohTelephoneNumber)}
            >
              <Text
                style={{ ...styles.detailsValueU, color: theme.colors.text }}
              >
                {valueOrDefault(aohTelephoneNumber, "")}
              </Text>
            </TouchableOpacity>
          )
        )}
      </View>
    </View>,
    <View style={styles.lineStyle} />,
    <View>
      <Text style={{ ...styles.detailsLabel, color: theme.colors.text }}>
        Email
      </Text>
      {(valueOrDefault(portAgent?.emailAddress, []) as string[]).map(
        (email: string, i: number) => (
          <TouchableOpacity
            testID="email-button"
            key={i}
            onPress={() => handleLink("mailto", email)}
          >
            <Text style={{ ...styles.detailsValueU, color: theme.colors.text }}>
              {valueOrDefault(email, "")}
            </Text>
          </TouchableOpacity>
        )
      )}
    </View>,
    <View>
      <Text style={{ ...styles.detailsLabel, color: theme.colors.text }}>
        Skype
      </Text>
      <TouchableOpacity
        testID="skype-button"
        onPress={() => handleLink("skype", portAgent?.skypeAddress)}
      >
        <Text style={{ ...styles.detailsValueU, color: theme.colors.text }}>
          {valueOrDefault(portAgent?.skypeAddress, "")}
        </Text>
      </TouchableOpacity>
    </View>,
    <View style={styles.lineStyle} />,
    <WebsiteView
      website={valueOrDefault(portAgent?.website, []) as string[]}
    />,
    <View>
      <Text style={{ ...styles.detailsLabel, color: theme.colors.text }}>
        Address
      </Text>
      <TouchableOpacity
        testID="address-button"
        onPress={() => handleLink("url", portAgent?.googleMapLink)}
      >
        <Text style={{ ...styles.detailsValueU, color: theme.colors.text }}>
          {valueOrDefault(portAgent?.address, "")}
        </Text>
      </TouchableOpacity>
    </View>,
    <View style={styles.endSpace} />,
  ];
};

const PortAgentContactDetails = (props: Props) => {
  const { portAgentDetails } = props;
  const theme = useTheme();
  const agentDetails = renderPortAgentDetails(theme, portAgentDetails);

  return (
    <ScrollView style={styles.root}>
      {agentDetails.map((column, i) => (
        <View key={i}>{column}</View>
      ))}
    </ScrollView>
  );
};

export default PortAgentContactDetails;
