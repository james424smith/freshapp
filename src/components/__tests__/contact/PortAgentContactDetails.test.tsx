import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react-native";
import PortAgentContactDetails from "../../contact/PortAgentContactDetails";
import * as HandleLink from "../../../common/handleLink";
import { portAgent } from "../../../../__mocks__/fake-data/contactApi.json";

describe("PortAgentContactDetails snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render PortAgentContactDetails correctly as per the ui made and  photoSmall is defined", () => {
    const { toJSON } = render(
      <PortAgentContactDetails portAgentDetails={portAgent} />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});

describe("should test fireEvents", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });
  it("should call handleLink when directTelephoneNumber-button is pressed ", () => {
    const { getByTestId } = render(
      <PortAgentContactDetails portAgentDetails={portAgent} />
    );

    const spy = jest.spyOn(HandleLink, "handleLink").mockImplementation();

    fireEvent.press(getByTestId("directTelephoneNumber-button"));

    expect(HandleLink.handleLink).toHaveBeenCalledWith(
      "tel",
      portAgent.directTelephoneNumbers[0]
    );
    spy.mockRestore();
  });
  it("should call handleLink when aohTelephoneNumber-button is pressed ", () => {
    const { getByTestId } = render(
      <PortAgentContactDetails portAgentDetails={portAgent} />
    );

    const spy = jest.spyOn(HandleLink, "handleLink").mockImplementation();

    fireEvent.press(getByTestId("aohTelephoneNumber-button"));

    expect(HandleLink.handleLink).toHaveBeenCalledWith(
      "tel",
      portAgent.aohTelephoneNumbers[0]
    );
    spy.mockRestore();
  });
  it("should call handleLink when email-button is pressed ", () => {
    const { getByTestId } = render(
      <PortAgentContactDetails portAgentDetails={portAgent} />
    );

    const spy = jest.spyOn(HandleLink, "handleLink").mockImplementation();

    fireEvent.press(getByTestId("email-button"));

    expect(HandleLink.handleLink).toHaveBeenCalledWith(
      "mailto",
      portAgent.emailAddress[0]
    );
    spy.mockRestore();
  });
  it("should call handleLink when skype-button is pressed ", () => {
    const { getByTestId } = render(
      <PortAgentContactDetails portAgentDetails={portAgent} />
    );

    const spy = jest.spyOn(HandleLink, "handleLink").mockImplementation();

    fireEvent.press(getByTestId("skype-button"));

    expect(HandleLink.handleLink).toHaveBeenCalledWith(
      "skype",
      portAgent.skypeAddress
    );
    spy.mockRestore();
  });
  it("should call handleLink when website-button is pressed ", () => {
    const { getByTestId } = render(
      <PortAgentContactDetails portAgentDetails={portAgent} />
    );

    const spy = jest.spyOn(HandleLink, "handleLink").mockImplementation();

    fireEvent.press(getByTestId("website-button"));

    expect(HandleLink.handleLink).toHaveBeenCalledWith(
      "url",
      portAgent.website[0]
    );
    spy.mockRestore();
  });
  it("should call handleLink when address-button is pressed ", () => {
    const { getByTestId } = render(
      <PortAgentContactDetails portAgentDetails={portAgent} />
    );

    const spy = jest.spyOn(HandleLink, "handleLink").mockImplementation();

    fireEvent.press(getByTestId("address-button"));

    expect(HandleLink.handleLink).toHaveBeenCalledWith(
      "url",
      portAgent.googleMapLink
    );
    spy.mockRestore();
  });
});
