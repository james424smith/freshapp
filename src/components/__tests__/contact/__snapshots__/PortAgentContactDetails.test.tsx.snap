// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortAgentContactDetails snapshot test should render PortAgentContactDetails correctly as per the ui made and  photoSmall is defined 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 22.5,
                "marginHorizontal": 60,
                "marginTop": 60,
              },
            ]
          }
        >
          Port Agent Name
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 26.5,
                "fontWeight": "bold",
                "marginHorizontal": 60,
                "marginTop": 2.5,
              },
            ]
          }
        >
          Inchcape Shipping Services
        </Text>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 22.5,
                  "marginHorizontal": 60,
                  "marginTop": 60,
                },
              ]
            }
          >
            Direct Phone Numbers
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="directTelephoneNumber-button"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                    "marginHorizontal": 60,
                    "marginTop": 2.5,
                    "textDecorationLine": "underline",
                  },
                ]
              }
            >
              34-93-4813960
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 22.5,
                  "marginHorizontal": 60,
                  "marginTop": 60,
                },
              ]
            }
          >
            AOH Phone Numbers
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="aohTelephoneNumber-button"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#2A2E43",
                    "fontFamily": "Roboto-Regular",
                  },
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                    "marginHorizontal": 60,
                    "marginTop": 2.5,
                    "textDecorationLine": "underline",
                  },
                ]
              }
            >
              34-93-4813960
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "borderBottomColor": "#78849E",
            "borderBottomWidth": 0.3,
            "marginTop": 60,
          }
        }
      />
    </View>
    <View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 22.5,
                "marginHorizontal": 60,
                "marginTop": 60,
              },
            ]
          }
        >
          Email
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="email-button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "marginHorizontal": 60,
                  "marginTop": 2.5,
                  "textDecorationLine": "underline",
                },
              ]
            }
          >
            consignacion.bcn@berge-m.es
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 22.5,
                "marginHorizontal": 60,
                "marginTop": 60,
              },
            ]
          }
        >
          Skype
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="skype-button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "marginHorizontal": 60,
                  "marginTop": 2.5,
                  "textDecorationLine": "underline",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "borderBottomColor": "#78849E",
            "borderBottomWidth": 0.3,
            "marginTop": 60,
          }
        }
      />
    </View>
    <View>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="website-button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 22.5,
                  "marginHorizontal": 60,
                  "marginTop": 60,
                },
              ]
            }
          >
            Web
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "marginHorizontal": 60,
                  "marginTop": 2.5,
                  "textDecorationLine": "underline",
                },
              ]
            }
          >
            1234
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 22.5,
                "marginHorizontal": 60,
                "marginTop": 60,
              },
            ]
          }
        >
          Address
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="address-button"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "marginHorizontal": 60,
                  "marginTop": 2.5,
                  "textDecorationLine": "underline",
                },
              ]
            }
          >
            Avda. Drassanes 6-8, 18a Planta Ed. Colon, 8001, Barcelona
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "marginBottom": "30%",
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
